### PR TITLE
Introduce ADR on project naming

### PR DESCRIPTION
**Pull request by giorgiosironi**
Summarizes https://docs.google.com/document/d/1ayJlohp2-s49c1_oypDE1bAfgYa5shR0I1GM6KRzDNA/edit after ad-hoc and Libero sync meetings.

### CODE REVIEW

**Review by BlueReZZ — APPROVED** [review]
Covers the discussion and removes the ambiguity around whether this promotes/forces/ignores reuse

**Review by NuclearRedeye — APPROVED** [review]

File: doc/adr/0006-project-naming.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Libero projects may span a single product or possibly more, but they live in a single namespace of the [Libero Github organization](https://github.com/libero). Hence the projects need to:
+
+- be attributable to the originating product
+- avoid clashes in naming (e.g. many `article-store` projects)
+
+## Decisions
+
+- Name repositories that cover multiple products without prefixes (e.g. `infrastructure`, `community`)
+- Name internal and/or product-specific repositories with prefix for specific product (e.g. `reviewer-submission`, `my-product-storybook`)
+- Name generic repositories with no prefix (e.g. `event-bus`)
```
Comment (davidcmoulton) [inline doc/adr/0006-project-naming.md]: What's the difference between "repositories that cover multiple products" (line 18) and "generic repositories"? Does it matter that these types are indistinguishable through this naming convention?

Comment (giorgiosironi) [reply doc/adr/0006-project-naming.md]: I understood the former as repositories that are not built into the product (`infrastructure` and `community` as examples), whereas the latter as reusable blocks of the products that may be shared. Perhaps it's not an important distiction, or it needs better naming.

Comment (thewilkybarkid) [reply doc/adr/0006-project-naming.md]: `event-bus` seems a bad example.

Comment (giorgiosironi) [reply doc/adr/0006-project-naming.md]: Agreed; do we have an existing example that better represents the category?

Comment (thewilkybarkid) [reply doc/adr/0006-project-naming.md]: Only archived things like https://github.com/libero/media-type I think.

Comment (giorgiosironi) [reply doc/adr/0006-project-naming.md]: Will work with that

**Comment by giorgiosironi** [comment]
The wording on generic repositories is probably as good as it gets according to the Libero Sync meeting, so not editing it any more.

**Review by diversemix — APPROVED** [review]
I would suggest the title be more explicit .... it's not about "project" naming - more about "repository, container and npm package" naming.

**Comment by giorgiosironi** [comment]
I mean `project` as a generic term to cover that but also similar artifacts that will come up as derivatives (e.h. Helm charts); any other generic term?
In my mind Docker/NPM and similar are derived from the repository name (more or less some suffixes if needed).

**Pull request merged**